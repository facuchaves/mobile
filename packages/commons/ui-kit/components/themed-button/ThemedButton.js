import PropTypes from 'prop-types'
import styled from 'styled-components'

const ThemedButton = styled.button`
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border-radius: 3px;

  /* Color the border and text with theme.main */
  background: ${props => props.theme.colors.primary.normal};
  color: ${props => props.theme.colors.primary.darken};
  border: 2px solid ${props => props.theme.colors.primary.lighten};
`

ThemedButton.propTypes = {
  children: PropTypes.node,
}

ThemedButton.defaultProps = {
  children: null,
}

export default ThemedButton
