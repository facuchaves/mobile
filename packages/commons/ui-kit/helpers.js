/* eslint-disable radix */
/* eslint-disable no-param-reassign */
/* eslint-disable prefer-const */
/* eslint-disable func-names */
/* eslint-disable import/prefer-default-export */

/* Resta el porcentaje indicado a un color (RR, GG o BB) hexadecimal para oscurecerlo */
const subtractLight = function(color, amount) {
  let cc = parseInt(color, 16) - amount
  let c = cc < 0 ? 0 : cc
  c = c.toString(16).length > 1 ? c.toString(16) : `0${c.toString(16)}`
  return c
}

/* Oscurece un color hexadecimal de 6 caracteres #RRGGBB segun el porcentaje indicado */
export const darken = (color, amount) => {
  color = color.indexOf('#') >= 0 ? color.substring(1, color.length) : color
  amount = parseInt((255 * amount) / 100)
  const partOne = subtractLight(color.substring(0, 2), amount)
  const partTwo = subtractLight(color.substring(2, 4), amount)
  const partThree = subtractLight(color.substring(4, 6), amount)
  return `#${partOne}${partTwo}${partThree}`
}

/* Suma el porcentaje indicado a un color (RR, GG o BB) hexadecimal para aclararlo */
const addLight = function(color, amount) {
  let cc = parseInt(color, 16) + amount
  let c = cc > 255 ? 255 : cc
  c = c.toString(16).length > 1 ? c.toString(16) : `0${c.toString(16)}`
  return c
}

/* Aclara un color hexadecimal de 6 caracteres #RRGGBB segun el porcentaje indicado */
export const lighten = (color, amount) => {
  color = color.indexOf('#') >= 0 ? color.substring(1, color.length) : color
  amount = parseInt((255 * amount) / 100)
  const partOne = addLight(color.substring(0, 2), amount)
  const partTwo = addLight(color.substring(2, 4), amount)
  const partThree = addLight(color.substring(4, 6), amount)
  return `#${partOne}${partTwo}${partThree}`
}
