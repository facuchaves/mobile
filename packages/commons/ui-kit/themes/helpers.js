import React from 'react'
import PropTypes from 'prop-types'
// eslint-disable-next-line import/no-extraneous-dependencies
import { map } from 'lodash'

import { themes, PortalTheme } from '.'

export const forEachTheme = componentCase => {
  return (
    <>
      {map(themes, (theme, portal) => (
        <PortalTheme key={portal} portal={portal}>
          {componentCase({ theme })}
        </PortalTheme>
      ))}
    </>
  )
}

export const ForEachTheme = ({ children }) => {
  const componentCase = typeof children === 'function' ? children : () => children
  return forEachTheme(componentCase)
}

ForEachTheme.propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
}
