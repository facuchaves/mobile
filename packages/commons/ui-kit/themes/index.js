import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'
import { SiteIds } from '@navent-jobs/config'

import { GlobalStyles } from '../components'
import zonajobs from './skins/zonajobs'
import bumeran from './skins/bumeran'
import konzerta from './skins/konzerta'
import multitrabajos from './skins/multitrabajos'
import laborum from './skins/laborum'

export const themes = {
  zonajobs,
  bumeran,
  konzerta,
  multitrabajos,
  laborum,
}

const portalsBySiteId = {
  [SiteIds.BMAR]: 'bumeran',
  [SiteIds.BMCL]: 'laborum',
  [SiteIds.BMEC]: 'multitrabajos',
  [SiteIds.BMMX]: 'bumeran',
  [SiteIds.BMPA]: 'konzerta',
  [SiteIds.BMPE]: 'bumeran',
  [SiteIds.BMVE]: 'bumeran',
  [SiteIds.ZJAR]: 'zonajobs',
}

// Si no se encuentra la variable Site Id se usara los estilos de bumeran como default
const portalEnv = portalsBySiteId[process.env.REACT_APP_SITE_ID] || portalsBySiteId.BMAR

// const ThemeLoger = () => {
//   const themeContext = useContext(ThemeContext)
//   console.log('Theme', themeContext)
//   return null
// }

if (!portalEnv) {
  console.warn(`REACT_APP_SITE_ID env variable MUST be defined and one of ${JSON.stringify(Object.values(SiteIds))}`)
}
// TODO: we could remove the other themes before building based on `portalEnv`

export const PortalTheme = ({ portal, children }) => {
  return (
    <ThemeProvider theme={themes[portal]}>
      <>
        <GlobalStyles />
        {children}
      </>
    </ThemeProvider>
  )
}

PortalTheme.propTypes = {
  portal: PropTypes.oneOf(Object.keys(themes)),
  children: PropTypes.node.isRequired,
}

PortalTheme.defaultProps = {
  portal: portalEnv,
}
