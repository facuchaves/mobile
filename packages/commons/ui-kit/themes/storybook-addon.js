/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/forbid-prop-types */
import React, { Component, useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import addons from '@storybook/addons'

import { themes, PortalTheme } from '.'
import { Button } from '../components'

const ADDON_ID = '@navent/ui-kit-theme'
const PANEL_ID = `${ADDON_ID}/panel`
const THEME_CHANGED = `${ADDON_ID}/theme-changed`

/**
 * `ThemePanel` es el componente encargado de mostrar los themes y dejar seleccionarlos
 * Es agregado como panel en StoryBook al registrar el addon
 */
class ThemePanel extends Component {
  selectTheme = themeId => {
    const { api } = this.props
    api.getChannel().emit(THEME_CHANGED, { themeId })
  }

  render() {
    const { active } = this.props
    if (!active) {
      return null
    }

    return (
      <div style={{ padding: 10 }}>
        {Object.keys(themes).map(themeId => (
          <PortalTheme portal={themeId} key={themeId}>
            <Button key={themeId} variant="primary" onClick={() => this.selectTheme(themeId)}>
              {themeId}
            </Button>
          </PortalTheme>
        ))}
      </div>
    )
  }
}

ThemePanel.propTypes = {
  api: PropTypes.any.isRequired,
  active: PropTypes.bool.isRequired,
}

/**
 * `WithPortalTheme` es el componente encargado de escuchar el evento de cambio de tema
 * y reaccionar ante el mismo. (usa hooks)
 * Wrappea la story con el `PortalTheme` usando el `theme` que corresponda
 */
const WithPortalTheme = ({ story, context }) => {
  const [portal, setPortal] = useState(process.env.STORYBOOK_PORTAL || 'bumeran')

  useEffect(() => {
    const handleThemeChange = ({ themeId }) => setPortal(themeId)
    addons.getChannel().on(THEME_CHANGED, handleThemeChange)

    return () => addons.getChannel().removeListener(THEME_CHANGED, handleThemeChange)
  }, [portal])

  return <PortalTheme portal={portal}>{story(context)}</PortalTheme>
}

WithPortalTheme.propTypes = {
  story: PropTypes.func.isRequired,
  context: PropTypes.object,
}

WithPortalTheme.defaultProps = {
  context: null,
}

/**
 * `withPortalTheme` es el _decorator_ que exponemos para registrar en StoryBook
 */
export const withPortalTheme = () => (story, context) => {
  return <WithPortalTheme story={story} context={context} />
}

addons.register(ADDON_ID, api => {
  // eslint-disable-next-line react/prop-types
  const render = ({ active }) => <ThemePanel api={api} active={active} />
  const title = 'Themes'

  addons.addPanel(PANEL_ID, {
    title,
    render,
  })
})
