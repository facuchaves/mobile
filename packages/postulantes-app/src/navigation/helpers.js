/* eslint-disable global-require */
/* eslint-disable import/prefer-default-export */
import { Navigation } from 'react-native-navigation'
import NetInfo from '@react-native-community/netinfo'
import ScreenIds from '../constants/ScreenIds'
import DefaultTheme from '../themes/DefaultTheme'
import i18n from '../i18n'

let currentScreen
let currentTabIndex
let prevTabIndex

export function setCurrentScreen(screen) {
  currentScreen = screen
}

export function getCurrentScreen() {
  return currentScreen
}

export const showOverlay = (screenId, passProps = {}, interceptTouchOutside = false) => {
  const backgroundColor = 'white'

  Navigation.showOverlay({
    component: {
      name: screenId,
      passProps,
      options: {
        overlay: {
          interceptTouchOutside,
        },
        layout: {
          backgroundColor,
        },
      },
    },
  })
}

export function dismissOverlay(componentId) {
  return Navigation.dismissOverlay(componentId)
}

export function getCurrentTabIndex() {
  return currentTabIndex
}

export function setCurrentTabIndex(tabIndex) {
  currentTabIndex = tabIndex
}

export function setPrevTabIndex(tabIndex) {
  prevTabIndex = tabIndex
}

export function getPrevTabIndex() {
  return prevTabIndex
}

export function pushScreen(componentId, screenName, passProps) {
  Navigation.push(componentId, {
    component: {
      name: screenName,
      passProps,
    },
  })
}

export const showModal = async (nameScreen, passProps) => {
  const connection = await NetInfo.fetch()
  const nameModal = connection.isInternetReachable ? nameScreen : ScreenIds.BOOT_ERROR

  Navigation.showModal({
    stack: {
      children: [
        {
          component: {
            name: ScreenIds[nameModal],
            passProps,
            options: {
              animations: {
                setStackRoot: {
                  enabled: true,
                },
              },
              topBar: {
                drawBehind: true,
                visible: false,
                animate: false,
              },
              layout: {
                orientation: ['portrait'],
              },
            },
          },
        },
      ],
    },
  })
}
export const changeBadge = totalUnread => {
  let badgeValue = ''
  if (totalUnread > 0) {
    badgeValue = totalUnread > 10 ? '10+' : `${totalUnread}`
  }

  Navigation.mergeOptions('Mensajes', {
    bottomTab: {
      badge: badgeValue,
    },
  })
}

export const dismissModal = componentId => {
  Navigation.dismissModal(componentId)
}

export const startSingleScreenApp = (initialScreen = ScreenIds.ONBOARDING) => {
  Navigation.setRoot({
    root: {
      component: {
        name: initialScreen,

        options: {
          layout: {
            orientation: ['portrait'],
          },
        },
      },
    },
  })
}

export const setDefaultNavigationOptions = theme => {
  Navigation.setDefaultOptions({
    layout: {
      orientation: ['portrait'],
      backgroundColor: theme.colors.background,
    },
  })
}

const getTabTintOptions = (theme, screen, text, icon, applyTint) => {
  if (applyTint) {
    if (ScreenIds.MENSAJES === screen) {
      return {
        id: text,
        text,
        iconColor: theme.colors.bottomTabDisabled,
        selectedIconColor: theme.colors.accent,
        textColor: theme.colors.bottomTabDisabled,
        selectedTextColor: theme.colors.accent,
        fontFamily: 'Helvetica',
        selectedFontSize: 14,
        fontSize: 14,
        icon,
        // badge: numberMenssage,
        badgeColor: 'red',
        titleDisplayMode: 'alwaysHide',
      }
    }
    return {
      id: text,
      text,
      iconColor: theme.colors.bottomTabDisabled,
      selectedIconColor: theme.colors.accent,
      textColor: theme.colors.bottomTabDisabled,
      selectedTextColor: theme.colors.accent,
      fontFamily: 'Helvetica',
      selectedFontSize: 14,
      fontSize: 14,
      icon,
      titleDisplayMode: '',
    }
  }

  return { icon }
}

const getBottomTab = (theme, screen, text, icon, applyTint = true) => {
  return {
    stack: {
      id: text,
      children: [
        {
          component: {
            name: screen,
            passProps: {
              text,
            },
            options: {
              bottomTab: getTabTintOptions(theme, screen, text, icon, applyTint),
              topBar: {
                drawBehind: true,
                visible: false,
                animate: false,
              },
              layout: {
                orientation: ['portrait'],
              },
              bottomTabs: {
                titleDisplayMode: 'alwaysShow',
                backgroundColor: theme.colors.primary,
                animate: false,
                visible: true,
                hideShadow: true,
                elevation: 10,
              },
            },
          },
        },
      ],
    },
  }
}

export const startLoginTabBasedApp = theme => {
  Navigation.setRoot({
    root: {
      bottomTabs: {
        id: 'bottom-tabs',
        children: [
          getBottomTab(theme, ScreenIds.HOME, i18n.t('tabs:home'), require('../images/icn_search.png')),
          getBottomTab(
            theme,
            ScreenIds.MIS_POSTULACIONES,
            i18n.t('tabs:mis_postulaciones'),
            require('../images/icn_post.png')
          ),
          getBottomTab(theme, ScreenIds.CURRICULUM, i18n.t('tabs:curriculum'), require('../images/icn_cv.png')),
          getBottomTab(theme, ScreenIds.MENSAJES, i18n.t('tabs:mensajes'), require('../images/icn_chat.png')),
        ],
        options: {
          bottomTabs: {
            animate: true,
            visible: true,
            hideShadow: true,
            titleDisplayMode: 'alwaysHide',
            currentTabIndex: 0,
            backgroundColor: DefaultTheme.colors.primary,
            elevation: 10,
          },
          bottomTab: {},
        },
      },
    },
  })
}

export const startTabBasedApp = () => {
  Navigation.setRoot({
    root: {
      component: {
        name: ScreenIds.PRE_LOGIN,
      },
    },
  })
}
