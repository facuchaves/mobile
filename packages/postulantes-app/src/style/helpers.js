import { useMemo, useRef, useCallback } from 'react'
import { Dimensions, Animated } from 'react-native'

import DefaultTheme from '../themes/DefaultTheme'

/**
 * Constants
 */

export const WINDOW_WIDTH = Dimensions.get('window').width
export const WINDOW_HEIGHT = Dimensions.get('window').height

const PORTRAIT_BASE_WIDTH = 375
const PORTRAIT_BASE_HEIGHT = 667

const portraitDeviceHeight = Math.max(WINDOW_WIDTH, WINDOW_HEIGHT)
const portraitDeviceWidth = Math.min(WINDOW_WIDTH, WINDOW_HEIGHT)

function vh(value) {
  return (portraitDeviceHeight * value) / PORTRAIT_BASE_HEIGHT
}

function vw(value) {
  return (portraitDeviceWidth * value) / PORTRAIT_BASE_WIDTH
}

const boxShadowStyle = {
  elevation: 3,
  shadowColor: DefaultTheme.colors.shadow,
  shadowOpacity: 1,
  shadowRadius: vw(3),
  shadowOffset: { width: 0, height: vw(5) },
  borderColor: 'transparent',
}

const boxStyle = {
  backgroundColor: DefaultTheme.colors.surface,
  shadowColor: DefaultTheme.colors.shadow,
  shadowOffset: { width: vw(1), height: vw(1) },
  shadowOpacity: 0.3,
  elevation: 2,
  borderRadius: vw(5),
  marginHorizontal: vw(16),
  marginVertical: vw(6),
}

const notificationBoxStyle = {
  backgroundColor: DefaultTheme.colors.surface,
  shadowColor: DefaultTheme.colors.shadow,
  shadowOffset: { width: vw(1), height: vw(1) },
  shadowOpacity: 0.3,
  elevation: 2,
  borderRadius: vw(5),
  marginVertical: vw(6),
}

/*
 * Hooks
 */

const useAnimatedStyles = initialOpacity => {
  const animatedOpacity = useRef(new Animated.Value(initialOpacity)).current

  const fadeIn = useCallback(() => {
    Animated.timing(animatedOpacity, {
      toValue: 1,
      duration: 500,
      useNativeDriver: true,
    }).start()
  }, [animatedOpacity])

  const fadeOut = useCallback(() => {
    Animated.timing(animatedOpacity, {
      toValue: 0,
      duration: 500,
      useNativeDriver: true,
    }).start()
  }, [animatedOpacity])

  const animatedStyles = useMemo(() => ({ opacity: animatedOpacity }), [animatedOpacity])

  return { animatedStyles, fadeIn, fadeOut }
}

/**
 * Constants
 */
export {
  portraitDeviceHeight,
  portraitDeviceWidth,
  vh,
  vw,
  boxShadowStyle,
  boxStyle,
  notificationBoxStyle,
  useAnimatedStyles,
}
