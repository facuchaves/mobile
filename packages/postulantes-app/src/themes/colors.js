export const transparent = 'transparent'
export const white = '#FFFFFF'
export const black = 'rgb(36, 40, 44)'
export const green = '#3CC431'
export const whatsAppGreen = '#1AD13F'
export const blue = '#0582F0'
export const ultraLightBlue = '#E4F0FC'
export const lightGrey10 = '#FBFBFB'
export const lightGrey20 = '#929292'
export const darkGrey10 = '#494949'
export const darkGrey20 = '#6A6D6E'
export const grey10 = '#3A3A44'
export const grey20 = '#BABABA'
export const purple10 = '#9E3DFF'
export const purple20 = '#7227BD'

/*
 * Enum
 */

// DEPRECATED - use name exported colors above instead, always from the theme
const Colors = Object.freeze({
  TRANSPARENT: 'transparent',
  TRANSPARENT_OPACITY: '#00000088',
  WHITE: '#FFFFFF',
  BLACK: '#24282C',
  LIGHT_GREY: '#FBFBFB',
  DARK_GREY: '#494949',
  TEXT_LIGHT_GREY: '#929292',
  TEXT_DARK_GREY: '#6A6D6E',
  TITLE_GREY: '#3A3A44',
  ALT_GREY: '#BABABA',
  DISABLED: '#CECECE',
  PURPLE: '#8231C4',
  GREEN: '#3CC431',
  BLUE: '#0582F0',
})

/*
 * Exports
 */

export default Colors
