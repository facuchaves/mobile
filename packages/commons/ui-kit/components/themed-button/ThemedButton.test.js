import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import ThemedButton from './ThemedButton'
import { forEachTheme } from '../../themes/helpers'

describe('ThemedButton snapshots', () => {
  it('default props', () => {
    const tree = renderer.create(forEachTheme(({ theme }) => <ThemedButton>{theme.portal}</ThemedButton>)).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
