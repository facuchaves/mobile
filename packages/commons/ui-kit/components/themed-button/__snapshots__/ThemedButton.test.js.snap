// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedButton snapshots default props 1`] = `
Array [
  .c0 {
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border-radius: 3px;
  color: royalblue;
  border: 2px solid royalblue;
}

<button
    className="c0"
  >
    zonajobs
  </button>,
  .c0 {
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border-radius: 3px;
  color: pink;
  border: 2px solid pink;
}

<button
    className="c0"
  >
    bumeran
  </button>,
]
`;
